### PR TITLE
Add named and typed httpclient example to the IHttpClientFactory with .NET page

### PR DESCRIPTION
**Pull request by peter-csala**
## Summary

The [Use the IHttpClientFactory / IHttpClientFactory with .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory) page shows how to use named or typed `HttpClient`s. 

But it does not even mention **named and typed** client.

The main aim of this pull request is to show how to register and use named and typed clients.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/extensions/httpclient-factory.md](https://github.com/dotnet/docs/blob/669bc221bbcaf2186137e622034120d1346e0245/docs/core/extensions/httpclient-factory.md) | [IHttpClientFactory with .NET](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory?branch=pr-en-us-40359) |

### CODE REVIEW

**Comment by peter-csala** [comment]
Hey David (@IEvangelist), could you please review this PR?

File: docs/core/extensions/httpclient-factory.md [hunk 1]
```diff
@@ -127,6 +128,48 @@ The typed client is registered as transient with DI. In the preceding code, `Add
...
+
+Named clients and typed clients have their own strengths and weaknesses. There is a way to combine these two client types to get the best both worlds.
+
+The primary use case is the following: Use the same typed client but against different domains. For example you have a primary and a secondary service and they expose the exact same functionality. That means you can use the same typed client to wrap the `HttpClient` usage to issue requests, process responses, and handle errors. The exact same code will be used but with different configurations (different base address, timeout, credentials, etc.).
+
+The following example uses the same `TodoService` typed client which was shown under the [typed clients](#typed-clients) section.
+
+First register the named and typed clients.
+
+:::code source="snippets/http/namedandtyped/Program.cs" range="1-25" highlight="9,14-15,19,23-24":::
```
Comment (IEvangelist) [inline docs/core/extensions/httpclient-factory.md:141]:
```suggestion
:::code source="snippets/http/namedandtyped/Program.cs" range="1-25" highlight="10,14-15,19,23-24":::
```

I think you wanted this line instead:

image

File: docs/core/extensions/httpclient-factory.md [hunk 2]
```diff
@@ -127,6 +128,48 @@ The typed client is registered as transient with DI. In the preceding code, `Add
...
+
+In the preceding code:
+
+- The first `AddHttpClient` call registers a `TodoService` typed client under the `primary` name. The underlying `HttpClient` points to the primary service and has a short timeout.
+- The second `AddHttpClient` call registers a `TodoService` typed client under the `secondary` name. The underlying `HttpClient` points to the secondary service and has a longer timeout.
+
+:::code source="snippets/http/namedandtyped/Program.cs" range="27-39" highlight="34-35,38-39":::
```
Comment (IEvangelist) [inline docs/core/extensions/httpclient-factory.md:148]: Highlights start from the beginning of the rendered `range`, and are 1-based I believe. I'm not sure what you're trying to highlight:

image

Comment (peter-csala) [reply docs/core/extensions/httpclient-factory.md]: Just fixed them :D

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**